refactor(tests): move helpers to new testing framework structure

vishalanarase opened the pull request:
<!-- 🎉🎉🎉 Thank you for the PR!!! 🎉🎉🎉 -->

# Changes

<!-- Describe your changes here- ideally you can get that description straight from your descriptive commit message(s)! -->

- Moved mock data creation from pipelinerunresolution_test.go to
  pkg/reconciler/testing/factory.go, including:
  - `p` → `pipeline`
  - `pts` → `pipelineTasks`
  - `newTaskRun()` and related setup helpers
- Moved status transformers to pkg/reconciler/testing/status_transformers.go:
  - `makeSucceed()`, `withCanceled()`, etc.
- Moved selectors to pkg/reconciler/testing/selectors.go:
  - `nopGetCustomRun`, etc.
- Renamed single-letter variables to descriptive names for clarity


# Submitter Checklist

As the author of this PR, please check off the items in this checklist:

- [ ] Has [Docs](https://github.com/tektoncd/community/blob/main/standards.md#docs) if any changes are user facing, including updates to minimum requirements e.g. Kubernetes version bumps
- [ ] Has [Tests](https://github.com/tektoncd/community/blob/main/standards.md#tests) included if any functionality added or changed
- [ ] [pre-commit](https://github.com/tektoncd/pipeline/blob/main/DEVELOPMENT.md#install-tools) Passed
- [ ] Follows the [commit message standard](https://github.com/tektoncd/community/blob/main/standards.md#commits)
- [ ] Meets the [Tekton contributor standards](https://github.com/tektoncd/community/blob/main/standards.md) (including functionality, content, code)
- [ ] Has a kind label. You can add one by adding a comment on this PR that contains `/kind <type>`. Valid types are bug, cleanup, design, documentation, feature, flake, misc, question, tep
- [ ] Release notes block below has been updated with any user facing changes (API changes, bug fixes, changes requiring upgrade notices or deprecation warnings). See some examples of [good release notes](https://github.com/tektoncd/community/blob/main/standards.md#good-release-notes).
- [ ] Release notes contains the string "action required" if the change requires additional action from users switching to the new release

# Release Notes

```release-note
NONE
```

Resolves https://github.com/tektoncd/pipeline/issues/8921

waveywaves commented:
/kind cleanup

waveywaves commented:
/kind misc

waveywaves commented on pkg/reconciler/testing/selectors.go:
```diff
@@ -0,0 +1,23 @@
+package testing
```
both `selectors.go` and `status_transformers.go` add exported helpers (`MakeStarted`, `MakeSucceeded`, `NopGetCustomRun`, `NopGetTask`, etc.) but none of them are actually called anywhere in this PR. The test files still use their own local `makeStarted`/`makeSucceeded` functions. If the plan is to migrate callers in a follow-up, good to note that in the PR description, otherwise these are dead code on arrival.

vishalanarase commented:
@waveywaves It's ready for review

vishalanarase commented:
/retest

vishalanarase committed:
test: reuse shared pipelinerun test helpers

Move mock data, status transformers, and selectors into
pkg/reconciler/testing/ and update pipelinerun resolution tests
to use the shared helpers.

Signed-off-by: Vishal Anarase <iamvishalanarase@gmail.com>

vdemeester reviewed (changes requested):
- I am not a huge fan of  `testhelpers.ExampleTaskRuns[0]`. As a *reader*, I have no idea what is the 1st (index 0) example taskrun, nor how many example taskruns there is. When it is in the same file it is "ok-ish", but in a different package is another story. Could we / Should we name those instead ?
- We may want to have small unit tests for the exported functions in the new `testing` package.
